build.sbt

```diff
@@ -1,4 +1,4 @@
-ThisBuild / tlBaseVersion := "0.0"
+ThisBuild / tlBaseVersion := "1.0"
```

Comment:
I'm dead serious about compatibility on this. No 0.0.x gang! Let's do our best to get it right 😇

Reply:
If we want to have long backward compatibility guarantees, I'm not sure open traits are a good idea.
You can't add a method to an open trait without breaking binary compatibility.
If you go from:
```scala
package foo

trait Trait {
  def bar: Int
}
```
to:
```scala
package foo

trait Trait {
  def bar: Int
  def foo: Int = defaultValue
}
```

You get this from MiMa:
```
Found 1 issue when checking against foo:before:1
 * interface foo.Foo does not have a correspondent in current version
   filter with: ProblemFilter.exclude[MissingClassProblem]("foo.Foo")
1 targets failed
after.mimaReportBinaryIssues Failed binary compatibility check! Found 1 potential problems
```

An API based on a class can instead be evolved in a binary backward compatible way:
```scala
class ScalaNativeRuntime private (
  _polling: Polling,
  _scheduler: Scheduler
) {
  def polling: Polling = _polling
  def scheduler: Scheduler = _scheduler

  private def copy(
    polling: Polling = this.polling,
    scheduler: Scheduler = this.scheduler
  ) = new ScalaNativeRuntime(polling, scheduler)

  def withPolling(polling: Polling): ScalaNativeRuntime =
    copy(polling = polling)
  def withScheduler(scheduler: Scheduler): ScalaNativeRuntime =
    copy(scheduler = scheduler)
}
```

You can add new fields as long as you provide a default (which can fail gracefully, for example).

Reply:
I had a doubt about binary compatibility with open traits. But I found that the problem exists only in Scala 2.11 (which we don't have to support). So it is fine :) Sorry

Reply:
I had a doubt about binary compatibility with open traits. But I found that the problem exists only in Scala 2.11 (which we don't have to support). So it is fine :) Sorry

Reply:
> I had a doubt about binary compatibility with open traits.

Yes, in Scala 2.11 you cannot add new methods to `trait`s with default implementations without breaking binary-compatibility. I don't expect us to add a method with a default implementation.

> But I found that the problem exists only in Scala 2.11 (which we don't have to support)

I think we can keep 2.11 as long as it's easy. If it gets in the way of bincompat, then perhaps we can drop it (or just break bincompat for it).